### PR TITLE
feat: Favorites

### LINKED ISSUE #820: Add Companies or People to favorite
When I click on the heart, it adds the card to a favorite section.

Image

Image

Image

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
Fixes: #820

Feedback is appreciated!

CC - @FelixMalfait 

PS: I had made a PR for this earlier, but its commit history got polluted so I made a new one.

https://github.com/twentyhq/twenty/assets/13139771/b2ecb14d-0628-4fc9-9180-ed7d2b9495d0

### CODE REVIEW

File: server/src/database/schema.prisma [hunk 1]
```diff
@@ -554,6 +557,21 @@ model Attachment {
...
+model Favorite {
+  id          String   @id @default(uuid())
+  workspaceId String?
+  /// @TypeGraphQL.omit(input: true, output: false)
+  person      Person?  @relation(fields: [personId], references: [id])
+  personId    String?
+  /// @TypeGraphQL.omit(input: true, output: false)
+  company     Company? @relation(fields: [companyId], references: [id])
+  companyId   String?
+  User        User?    @relation(fields: [userId], references: [id])
```
Comment (charlesBochet) [inline server/src/database/schema.prisma:569]: let's link favorites to a workspaceMember actually

**Review by charlesBochet — COMMENTED** [review]
Thank you, nice first shot. Left comments :)

File: front/src/modules/favorites/states/currentFavorites.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { atom } from 'recoil';
+
+import { GetFavoritesQuery } from '~/generated/graphql';
+
+export const currentFavorites = atom<GetFavoritesQuery['findFavorites'] | []>({
```
[the remark this answers is not in the document]
Comment (AdityaPimpalkar) [reply front/src/modules/favorites/states/currentFavorites.ts:5]: I didn't I wasn't using this hook anywhere in my code too. Thanks!

File: front/src/modules/favorites/hooks/useSetFavorites.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useEffect, useState } from 'react';
+
+import { GetFavoritesQuery, useGetFavoritesQuery } from '~/generated/graphql';
+
+export function useFavorites() {
+  const { data } = useGetFavoritesQuery();
+  const [favorites, setFavorites] = useState<GetFavoritesQuery>();
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (AdityaPimpalkar) [reply front/src/modules/favorites/hooks/useSetFavorites.ts:8]: You're right, I didn't notice I wasn't this hook anywhere too. Thanks!

**Comment by charlesBochet** [comment]
@AdityaPimpalkar Let me know if you need help on this one

File: front/src/modules/favorites/states/currentFavorites.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdityaPimpalkar) [reply front/src/modules/favorites/states/currentFavorites.ts:5]: I removed the `currentFavorites` state altogether, using the Apollo state directly. Thanks!

File: server/src/database/schema.prisma (hunk 1 above)
Comment (AdityaPimpalkar) [reply server/src/database/schema.prisma:569]: Do you want me to replace the `User` with `workspaceMember`?

Comment (charlesBochet) [reply server/src/database/schema.prisma:569]: yes! A User (email + password) can be tied to different workspaces through workspaceMember.
Favorites should be attached to worksapceMember

Comment (AdityaPimpalkar) [reply server/src/database/schema.prisma:569]: It's automatically adding the User relation on save, I don't have `format on save` turned on in my vscode too, any suggestions? Thanks

Comment (charlesBochet) [reply server/src/database/schema.prisma:572]: It's your VSCode Prisma extension, you should also update your User model upper in the file and move Favorite to workspaceMember

Comment (AdityaPimpalkar) [reply server/src/database/schema.prisma]: Done! Let me know If I missed something

File: front/src/modules/ui/chip/components/Chip.tsx [hunk 4]
```diff
@@ -112,6 +112,11 @@ const StyledLabel = styled.span`
   white-space: nowrap;
 `;
 
+const ContainerWrapper = styled.div`
```
Comment (charlesBochet) [inline front/src/modules/ui/chip/components/Chip.tsx:115]: We should dismiss these modifications

**Review by charlesBochet — COMMENTED** [review]
Last round of comment that are mostly cosmetic!
Thanks for the work!

**Comment by charlesBochet** [comment]
Also, you'll need to resolve conflicts you have (pull Twenty/main and merge it into your branch) or you can use Github interfaces if this is easier

**Review by charlesBochet — APPROVED** [review]
LGTM, you are missing the migration but I need to handle it anyway because we've introduced some breaking changes that we need to fix first

**Comment by charlesBochet** [comment]
It's merged, thank you!

**Pull request merged**